extensions/amp-link-rewriter/0.1/amp-link-rewriter.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Layout} from '../../../src/layout';
+import {LinkRewriter} from './link-rewriter';
+import {Priority} from '../../../src/service/navigation';
+import {Services} from '../../../src/services';
+import {getConfigOpts} from './config-options';
+import {getScopeElements} from './scope';
+
+export class AmpLinkRewriter extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {!../../../src/service/ampdoc-impl.AmpDoc} */
+    this.ampDoc_ = this.getAmpDoc();
+
+    /** @private {?./link-rewriter.LinkRewriter} */
+    this.rewriter_ = null;
+
+    /** @private {?Object} */
+    this.configOpts_ = {};
+
+    this.navigation_ = Services.navigationForDoc(this.ampDoc_);
+
+    /** @private {?Array<!Element>} */
+    this.listElements_ = null;
+  }
+
+  /** @override */
+  buildCallback() {
+    this.configOpts_ = getConfigOpts(this.element);
+
+    return this.ampDoc_.whenBodyAvailable()
+        .then(this.letsRockIt_.bind(this));
+  }
+
+  /**
+   * @private
+   */
+  letsRockIt_() {
+    this.rewriter_ = new LinkRewriter(
```

Comment:
I saw you calculate `this.configOpts_` again in `LinkRewriter`. Please see my comment below I think we can move `this.listElement_` initiation to within `LinkRewriter` class, so that you won't need to calculate twice.

Reply:
fixed